### ISSUE TITLE
New SE systems

### ISSUE THREAD

**Issue by letmoc** [TA2k/ioBroker.gruenbeck#61]
Hello, 
could the new SE systems be included?
I have an SE21
Thanks

**Comment by Totte23** [comment]
I am interested as well. Also got SE21....

**Comment by TA2k** [comment]
Please provide an account login and password to
tombox2020@gmail.com

**Comment by TA2k** [comment]
Please test v0.0.44

**Comment by Totte23** [comment]
> Please test v0.0.44

Connection is working! Haven‘t checked the values yet. Great work!

**Comment by NigiNagiNu** [comment]
Also recently got a SE18 installed. Adapter is 0.0.46 version. Adapter is green and general statistics get updated. But Stream folder stays empty and I only see {"type":6} entries every 5 to 10 seconds in the debug log. Anyone encountering the same?

**Comment by Totte23** [comment]
Also empty with my SE21. What is expected to be within Stream? I am just interested in the consumption of water and Salt…

**Comment by NigiNagiNu** [comment]
To my understanding, there should be live data under Stream. Basically at least the same information, the SE18 is showing on its display. But I general have the impression, that cloud connection by Grünbeck is quite buggy. My SE18 shows proper cloud connection, but I never can't get registration in app successful over point 9 (finishing cloud connection). Also live data in app is working only unreliably. Just configured a simple device watch in my Fritzbox on the device, to get a feel for its communication. Seems to send an alive every 2 mins. If cloud connection is working properly and app is connected successful, then device is sending for at least 10 mins after app connection.

**Comment by NigiNagiNu** [comment]
@Totte23: What happens on your end, if you enable the adapter? As soon as I have running the adapter for a while, live data provision in the app stops. But things live parameter changes still seem to work - e.g. triggering the regenration from UI still works. Can also be observed in Fritzbox device tracking. If connection works, as soon as I open the app, Grünbeck device seems to start data streaming. If live data provision stops, Grünbeck device only send its alive every two minutes, but doesn't do any data streaming any longer.

**Comment by Totte23** [comment]
> @Totte23: What happens on your end, if you enable the adapter? As soon as I have running the adapter for a while, live data provision in the app stops. But things live parameter changes still seem to work - e.g. triggering the regenration from UI still works. Can also be observed in Fritzbox device tracking. If connection works, as soon as I open the app, Grünbeck device seems to start data streaming. If live data provision stops, Grünbeck device only send its alive every two minutes, but doesn't do any data streaming any longer.

You are correct. Same on my side. What I do is following as that is all I require:
- do restart of adapter each nigth at 2:00 am => then new values are called
- look at the JSON under measurments that show consumption of water and salt 
- write new values to influxdb for the dates shown in the json

To be honest: water measurement is not very accurate. I use this in parallel to take values directly from the water meter.
https://github.com/jomjol/AI-on-the-edge-device
Quite high deviations within 1 month => higher than 1m³

**Comment by NigiNagiNu** [comment]
> > @Totte23: What happens on your end, if you enable the adapter? As soon as I have running the adapter for a while, live data provision in the app stops. But things live parameter changes still seem to work - e.g. triggering the regenration from UI still works. Can also be observed in Fritzbox device tracking. If connection works, as soon as I open the app, Grünbeck device seems to start data streaming. If live data provision stops, Grünbeck device only send its alive every two minutes, but doesn't do any data streaming any longer.
> 
> You are correct. Same on my side. What I do is following as that is all I require:
> 
> * do restart of adapter each nigth at 2:00 am => then new values are called
> * look at the JSON under measurments that show consumption of water and salt
> * write new values to influxdb for the dates shown in the json
> 
> To be honest: water measurement is not very accurate. I use this in parallel to take values directly from the water meter. https://github.com/jomjol/AI-on-the-edge-device Quite high deviations within 1 month => higher than 1m³

Thanks for the confirmation. Sounds like I have to head into the same direction for the time being. Regeneration I can detect, as I have a measurement plug without relay connected. And our water meter had been replaced recently with a highly accurate ultrasonic one with wmbus (and as our town has individual AES keys on each, the key had been shared with us).

**Comment by linos22** [comment]
> [@Totte23](https://github.com/Totte23): What happens on your end, if you enable the adapter? As soon as I have running the adapter for a while, live data provision in the app stops. But things live parameter changes still seem to work - e.g. triggering the regenration from UI still works. Can also be observed in Fritzbox device tracking. If connection works, as soon as I open the app, Grünbeck device seems to start data streaming. If live data provision stops, Grünbeck device only send its alive every two minutes, but doesn't do any data streaming any longer.

Same for me that the live data in the app stops after enabling the adapter. 
Any fixes or tips here?

**Comment by TA2k** [comment]
How old is the device
Please try the GitHub version

**Comment by linos22** [comment]
> How old is the device Please try the GitHub version
2 days

**Comment by linos22** [comment]
> > How old is the device Please try the GitHub version
> > 2 days and I am using 0.0.46

Sorry, I thought I was using 0.0.46 based on the beta repository of iobroker, but this was not the case. It shows up as 0.0.46 but in reality it is 0.0.43. 
I installed 0.0.46 via GitHub and the Streamfolder is now filled with values. Let’s see if it works now

**Comment by linos22** [comment]
> > > How old is the device Please try the GitHub version
> > > 2 days and I am using 0.0.46
> 
> Sorry, I thought I was using 0.0.46 based on the beta repository of iobroker, but this was not the case. It shows up as 0.0.46 but in reality it is 0.0.43. I installed 0.0.46 via GitHub and the Streamfolder is now filled with values. Let’s see if it works now

Just to let you all know:
0.0.46 was fine to get all the objects in the Stream folder. But still after some time the updates to iobroker and also to the iOS App of Gruenbeck stopped. 
I assumed that there is some kind of limitation of requests, so I set the interval to 3600 seconds. Since then it runs smoothly

**Comment by winnieXY** [comment]
My Stream folder is empty with SE 21 - although all other information is available. Any hints what's wrong here?

With best regards
Patrick

**Comment by linos22** [comment]
> My Stream folder is empty with SE 21 - although all other information is available. Any hints what's wrong here?
> 
> With best regards Patrick

I had the same issue. Use the GitHub version instead of the Repository from IObroker

**Comment by winnieXY** [comment]
Indeed. Expert mode and install of 0.0.46 via npm doesn't work. Expert mode and install from github does work. Thanks!

**Comment by Ringo-66** [comment]
I'm using the SE21 as well,
Installed adapterversion is 0.0.46.
after (re)start I got this errors:

```
gruenbeck.0 | 2026-02-09 13:53:58.201 | info | terminating
-- | -- | -- | --
gruenbeck.0 | 2026-02-09 13:53:57.873 | error | Failed enter SD
gruenbeck.0 | 2026-02-09 13:53:57.873 | error | AxiosError: Request failed with status code 404
gruenbeck.0 | 2026-02-09 13:53:57.873 | error | https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices//realtime/enter?api-version=2024-05-02
gruenbeck.0 | 2026-02-09 13:53:57.849 | error | AxiosError: Request failed with status code 404
gruenbeck.0 | 2026-02-09 13:53:57.847 | error | AxiosError: Request failed with status code 404
gruenbeck.0 | 2026-02-09 13:53:57.839 | error | AxiosError: Request failed with status code 404
gruenbeck.0 | 2026-02-09 13:53:57.699 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
gruenbeck.0 | 2026-02-09 13:53:57.699 | info | terminating
gruenbeck.0 | 2026-02-09 13:53:57.699 | error | TypeError: Cannot read properties of null (reading 'close')
gruenbeck.0 | 2026-02-09 13:53:57.698 | error | unhandled promise rejection: undefined
gruenbeck.0 | 2026-02-09 13:53:57.698 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
gruenbeck.0 | 2026-02-09 13:53:57.698 | error | Error: The id "." is invalid. Ids are not allowed to end in "."
gruenbeck.0 | 2026-02-09 13:53:57.624 | error | TypeError: Cannot read properties of undefined (reading 'id')
gruenbeck.0 | 2026-02-09 13:53:57.624 | info | Found config index: '0'
gruenbeck.0 | 2026-02-09 13:53:57.624 | info | Filtered to 0 devices
gruenbeck.0 | 2026-02-09 13:53:57.624 | info | Found 1 devices
gruenbeck.0 | 2026-02-09 13:53:57.196 | info | Login successful
gruenbeck.0 | 2026-02-09 13:53:56.856 | info | Login step 2
gruenbeck.0 | 2026-02-09 13:53:56.359 | info | Login step 1
gruenbeck.0 | 2026-02-09 13:53:55.945 | info | Connection to SD Device with:****MAIL as logged in with account ****
gruenbeck.0 | 2026-02-09 13:53:55.935 | info | starting. Version 0.0.46 in /opt/iobroker/node_modules/iobroker.gruenbeck, node: v20.18.1, js-controller: 7.0.7

```
Whow do I get this up and running?

**Comment by TA2k** [comment]
Activate debug log to see which device was found
